File: capture-tui-go/internal/models/category.go
package models

import "time"

// Category represents a category for organizing entries
type Category struct {
	Name        string     `json:"name" yaml:"name"`
	DisplayName string     `json:"display_name" yaml:"display_name"`
	Count       int        `json:"count" yaml:"count"`
	CreatedAt   time.Time  `json:"created_at" yaml:"created_at"`
	LastEntry   *time.Time `json:"last_entry,omitempty" yaml:"last_entry,omitempty"`
	Description string     `json:"description" yaml:"description"`
}

// NewCategory creates a new category
func NewCategory(name, displayName, description string) *Category {
	return &Category{
		Name:        name,
		DisplayName: displayName,
		Count:       0,
		CreatedAt:   time.Now(),
		Description: description,
	}
}

// UpdateCount updates the entry count and last entry time
func (c *Category) UpdateCount(delta int) {
	c.Count += delta
	if c.Count < 0 {
		c.Count = 0
	}
	if delta > 0 {
		now := time.Now()
		c.LastEntry = &now
	}
}

// SetDisplayName sets the display name, defaulting to the category name
func (c *Category) SetDisplayName(displayName string) {
	if displayName == "" {
		c.DisplayName = c.Name
	} else {
		c.DisplayName = displayName
	}
}

File: capture/capture_tui/models/category.py
"""Category 数据模型"""

from dataclasses import dataclass, asdict
from datetime import datetime
from typing import Optional


@dataclass
class Category:
    """分类模型"""
    name: str
    display_name: str
    count: int = 0
    created_at: datetime = None
    last_entry: Optional[datetime] = None
    description: str = ""
    
    def __post_init__(self):
        if self.created_at is None:
            self.created_at = datetime.now()
    
    def to_dict(self) -> dict:
        return {
            "name": self.name,
            "display_name": self.display_name,
            "count": self.count,
            "created_at": self.created_at.isoformat() if self.created_at else None,
            "last_entry": self.last_entry.isoformat() if self.last_entry else None,
            "description": self.description
        }
    
    @classmethod
    def from_dict(cls, data: dict) -> "Category":
        return cls(
            name=data["name"],
            display_name=data.get("display_name", data["name"]),
            count=data.get("count", 0),
            created_at=datetime.fromisoformat(data["created_at"]) if data.get("created_at") else None,
            last_entry=datetime.fromisoformat(data["last_entry"]) if data.get("last_entry") else None,
            description=data.get("description", "")
        )
    
    def update_count(self, delta: int = 1):
        """更新条目计数"""
        self.count += delta
        if delta > 0:
            self.last_entry = datetime.now()
